argoverse/evaluation/eval_detection.py

```diff
@@ -0,0 +1,402 @@
+            detection and its ground truth assignment.
+
+        2. Average Scale Error: The average intersection over union (IoU) after the prediction
+            and assigned ground truth's pose has been aligned.
+        
+        3. Average Orientation Error: The average angular distance between the detection and
+            the assigned ground truth. We choose the smallest angle between the two different
+            headings when calculating the error.
+
+    Composite Scores
+
+        1. Composite Detection Score: The ranking metric for the detection leaderboard. This
+            is computed as the product of mAP with the sum of the complements of the true positive
+            errors (after normalization), i.e.,
+
+            Average Translation Measure (ATM): ATE / TP_THRESHOLD; 0 <= 1 - ATE / TP_THRESHOLD <= 1
+            Average Scaling Measure (ASM): 1 - ASE / 1;  0 <= 1 - ASE / 1 <= 1
+            Average Orientation Measure (AOM): 1 - AOE / PI; 0 <= 1 - AOE / PI <= 1
+
+            These (as well as AP) are averaged over each detection class to produce:
+
+            mAP, mATM, mASM, mAOM.
+
+            Lastly, the Composite Detection Score is computed as:
+
+            CDS = mAP * (mATE + mASE + mAOE); 0 <= mAP * (mATE + mASE + mAOE) <= 1.
+
+        ** In the case of no true positives under the specified threshold, the true positive measures
+            will assume their upper bounds of 1.0. respectively.
+
+Results:
+
+    The results are represented as a (C + 1, P) table, where C + 1 represents the number of evaluation classes
+    in addition to the mean statistics average across all classes, and P refers to the number of included statistics, 
+    e.g. AP, ATE, ASE, AOE, CDS by default.
+
+"""
+import argparse
+import logging
+import os
+from collections import defaultdict
+from dataclasses import Field, dataclass, field
+from multiprocessing import Pool
+from pathlib import Path
+from typing import DefaultDict, List, Tuple
+
+import matplotlib
+import numpy as np
+import pandas as pd
+from pandas.core import frame
+
+from argoverse.data_loading.object_classes import OBJ_CLASS_MAPPING_DICT
+from argoverse.data_loading.object_label_record import read_label
+from argoverse.evaluation.detection_utils import (
+    DistFnType,
+    FilterMetric,
+    SimFnType,
+    compute_affinity_matrix,
+    dist_fn,
+    filter_instances,
+    interp,
+    rank,
+)
+
+matplotlib.use("Agg")  # isort:skip
+
+
+import matplotlib.pyplot as plt  # isort:skip
+
+
+logger = logging.getLogger(__name__)
+
+
+TP_ERROR_NAMES: List[str] = ["ATE", "ASE", "AOE"]
+N_TP_ERRORS: int = len(TP_ERROR_NAMES)
+
+STATISTIC_NAMES: List[str] = ["AP"] + TP_ERROR_NAMES + ["CDS"]
+
+MAX_YAW_ERROR = np.pi
+
+MIN_AP: float = 0.0
+MAX_NORM_ATE: float = 1.0
+MAX_NORM_ASE: float = 1.0
+MAX_NORM_AOE: float = 1.0
+MIN_CDS: float = 0.0
+MEASURE_DEFAULT_VALUES: List[float] = [MIN_AP, MAX_NORM_ATE, MAX_NORM_ASE, MAX_NORM_AOE, MIN_CDS]
```

Comment:
thanks for updating this Ben. much more readable now. I'm not super certain about the term norm here -- i guess it works, do you think anything else would be a little more clear? maybe `MAX_AOE_VAL`? Not certain we can call a yaw angle a norm

Reply:
yeah, that's poor naming on my part. i guess we want to convey that this is the max error after normalizing it by the maximum value ... would saying something like `NORMALIZATION_CONST_AOE` be misleading?

Reply:
I see, ok that makes sense.
I like `MAX_NORMALIZED_AOE: float = 1.0`. yeah, it's a bit longer, but to me it's clear